compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -849,7 +883,6 @@ where
     }
 
     fn visit_ty(&mut self, ty: Ty<'tcx>) -> ControlFlow<Self::BreakTy> {
-        // Need to lazily normalize here in with `-Ztrait-solver=next-coherence`.
         let ty = match (self.lazily_normalize_ty)(ty) {
             Ok(ty) => ty,
```

Comment:
That does fix `orphan-check-projection-does-cover.rs#next` but it doesn't fix `orphan-check-projection-does-cover.rs#next` (should pass but fails). I told me that you were happy regardless of `does-cover` passing or failing but wasn't #100555 closed precisely because it lead to too much breakage in the ecosystem which seems to use the pattern `<T as Discard>::DoesntNormToSthContainingT`?